### PR TITLE
Feature/Remote Control - Radio and Climate Parameter Update

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
ATF Test Scripts to check  https://github.com/smartdevicelink/sdl_core/issues/2793

This PR is **ready** for review.

### Summary
This proposal changes the minimum index of HD radio sub-channels from 1 to 0.
In addition, we propose to add a new parameter climateEnable to ClimateControlData which will allow an application to power climate control on or off.

### ATF version
develop (checked on https://github.com/smartdevicelink/sdl_atf/commit/109d1cb39513ff0988be912597c4924e0f2387db)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/RC/CLIMATE_RADIO/OnInteriorVehicleData/017_Success_flow_with_climateEnable.lua [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+--[[ Local Variables ]]
+local mType = "CLIMATE"
+local paramValues = {
+  false,
+  true
+}
+
+--[[ Local Functions ]]
+local function notificationProcessedSuccessfully(pValue)
+  function commonRC.getAnotherModuleControlData()
```
Comment (JackLivio) [inline test_scripts/RC/CLIMATE_RADIO/OnInteriorVehicleData/017_Success_flow_with_climateEnable.lua]: I am not familiar with this syntax. Does this overwrite the definition of the common `getAnotherModuleControlData()`?

Comment (dboltovskyi) [reply test_scripts/RC/CLIMATE_RADIO/OnInteriorVehicleData/017_Success_flow_with_climateEnable.lua]: @JackLivio Correct. When `notificationProcessedSuccessfully()` function is executed implementation of `commonRC.getAnotherModuleControlData()` will be overwritten. And new code will be used till the end of the script.

**Comment by JackLivio** [comment]
Seeing a failure with this test `./test_scripts/RC/CLIMATE_RADIO/SetInteriorVehicleData/023_availableHDs_hdChannel_out_of_upper_bound.lua`

```
[13:12:16,168] SetInteriorVehicleData_set_incorrect_values_for_hdChannel_as_outOfLowerBoundValue     [FAIL] (10050 ms)
 Response to 5: ValidIf: 
payload.resultCode: expected: INVALID_DATA, actual value: GENERIC_ERROR
 HMI call RC.SetInteriorVehicleData: Times: The most allowed occurences boundary exceed
```

**Comment by JackLivio** [comment]
I believe that the test is sending a 0 as a lower out of bound value and the test is expecting it to fail. However the changes for this feature should allow 0 to be sent for an hdChannel.

**Comment by JackLivio** [comment]
Note i updated the syncMessage version of the apps in the config.lua to 6.0 before running the tests.

**Comment by dboltovskyi** [comment]
@JackLivio 
Thank you for this finding.
Unfortunately last commit `Update regarding changed min value for hdChannels` was made by mistake.
It changed min value for `hdChannel` parameter to 1 though it should be 0 as before.

Now this commit is removed and additionally PR is re-based onto HEAD of develop.
We also found inconsistency between MOBILE and HMI APIs regarding `hdChannel` parameter which is fixed in https://github.com/smartdevicelink/sdl_core/pull/2920/commits/009949655f9e34154a5a1ad75eb64a141fc9a44f

**Comment by JackLivio** [comment]
@dboltovskyi Thank you, re-running tests now.

**Pull request merged**